tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,408 @@
+
+        pods = self.run_oc_command(cmd="get pod", namespace="openshift-storage")
+        if self.params["ERRMSG"] in pods:
+            raise Exception("Can not get csi controller pod")
+
+        provisioning_name = "csi-cephfsplugin-provisioner"
+        if self.interface == constants.CEPHBLOCKPOOL:
+            provisioning_name = "csi-rbdplugin-provisioner"
+
+        for line in pods:
+            if provisioning_name in line:
+                self.params["logs"].append(line.split()[0])
+        log.info(f'The logs pods are : {self.params["logs"]}')
+
+    def run_fio_on_pod(self, file_size):
+        """
+        Args:
+        file_size (str): Size of file to write in MB, e.g. 200M or 50000M
+
+        """
+        file_name = self.pod_obj.name
+        log.info(f"Starting IO on the POD {self.pod_obj.name}")
+        # Going to run only write IO to fill the PVC for the before creating a clone
+        self.pod_obj.fillup_fs(size=file_size, fio_filename=file_name)
+
+        # Wait for fio to finish - non default timeout is needed for big pvc/clones
+        fio_result = self.pod_obj.get_fio_results(timeout=18000)
+        err_count = fio_result.get("jobs")[0].get("error")
+        assert err_count == 0, (
+            f"IO error on pod {self.pod_obj.name}. " f"FIO result: {fio_result}."
+        )
+        log.info("IO on the PVC Finished")
+
+        # Verify presence of the file on pvc
+        file_path = pod.get_file_path(self.pod_obj, file_name)
+        log.info(f"Actual file path on the pod is {file_path}.")
+        assert pod.check_file_existence(
+            self.pod_obj, file_path
+        ), f"File {file_name} does not exist"
+        log.info(f"File {file_name} exists in {self.pod_obj.name}.")
+
+    def create_clone(self, clone_num, clone_yaml):
+        """
+        Creating clone for pvc, measure the creation time
+
+        Args:
+            clone_num (int) the number of clones to create
+            clone_yaml : a template of clone yaml
+
+        Returns:
+            int: the creation time of the clone (in secs.)
+
+        """
+        log.info(f"Creating clone number {clone_num} for interface {self.interface}")
+
+        clone_name = f"pvc-clone-{clone_num}-"
+        clone_name += self.params["pvcname"].split("-")[-1]
+        clone_yaml["metadata"]["name"] = clone_name
+
+        fd, tmpfile = tempfile.mkstemp(suffix=".yaml", prefix="Clone")
+        log.info(f"Going to create {tmpfile}")
+        with open(tmpfile, "w") as f:
+            yaml.dump(clone_yaml, f, default_flow_style=False)
+        start_time = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        log.info(f"Clone yaml file is {clone_yaml}")
+        res = self.run_oc_command(f"create -f {tmpfile}", self.params["nspace"])
+        time.sleep(3)
+        if self.params["ERRMSG"] in res[0]:
+            raise Exception(f"Can not create clone : {res}")
+        # wait until clone is ready
+        timeout = 600
```

Comment:
Possible to add function similar to wait_for_resource_state ?

Reply:
> Possible to add function similar to wait_for_resource_state ?

Added the following function :  wait_for_clone_creation(self, clone_name, timeout=600)